### PR TITLE
Add 'data' evaluation as XPath in IntentCallout

### PR DESCRIPTION
**Pull request by wpride**
Small feature for Ismaïla and UNCDF. Allows the `data` field in `IntentCallouts` to be evaluated as an XPath. They need this to preload phone numbers for making calls.

Currently there's no way to set the `data` field in an `<intent>` block in HQ besides editing the Form XML directly so I don't think that there're likely to be existing users. However, this should continue to work gracefully if anyone was using this as a constant field. 

Not needed until February so can wait for the next release.

### CODE REVIEW

**Comment by amstone326** [comment]
@wpride Won't this break all existing usages of intent callouts in existing apps, since they've been configured to pass a string without quotes and expect it to be treated as a string? We've had a very similar problem to this before.

**Comment by amstone326** [comment]
@wpride was the last commit meant to address my concern? If so I'm not following how it does

**Comment by ctsims** [comment]
==Aliza I'm not following how this won't break existing callouts

Previously if you had 
```
<data>FOO</data>
```
that would have been parsed as a string and forwarded along.

Now you'd get a parse error, since that's not a string.

**Comment by wpride** [comment]
@ctsims @amstone326 Honestly not sure I'm still understanding your concern exactly, but added four tests demonstrating why I believe this is still safe. These cover:

1. Intent with XPath `data` attribute
2. Intent with no `data` attribute
3. Intent with empty ("") `data` attribute
4. Intent with hard coded `data` attribute

These intents still parse and behave fine. Am I missing something?

**Comment by amstone326** [comment]
(offlined briefly with Will but there are failing tests so I think that should help clarify)

**Comment by amstone326** [comment]
@wpride Can this be closed at this point or is it still needed by UNCDF?

**Comment by wpride** [comment]
@damagatchi retest this please

File: app/unit-tests/src/org/commcare/android/tests/formentry/IntentCalloutTests.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+                "jr://resource/commcare-apps/phone_call/profile.ccpr",
+                "test", "123");
+        ShadowEnvironment.setExternalStorageState(Environment.MEDIA_MOUNTED);
+    }
+
+    /**
+     * Test different behaviors for possibly grouped intent callout views
+     */
+    @Test
+    public void testIntentCalloutWithData() {
```
Comment (amstone326) [inline app/unit-tests/src/org/commcare/android/tests/formentry/IntentCalloutTests.java]: Can you rename this test to make it more obvious that this is the one testing the XPath format?

File: app/unit-tests/resources/commcare-apps/phone_call/default/app_strings.txt [hunk 2]
```diff
@@ -0,0 +1,26 @@
+app.display.name=1Scratchwork
```
Comment (amstone326) [inline app/unit-tests/resources/commcare-apps/phone_call/default/app_strings.txt]: Not positive but I think you can get rid of the app_strings resources and have the tests still work fine

File: app/unit-tests/resources/commcare-apps/phone_call/media_suite.xml [hunk 3]
```diff
@@ -0,0 +1,2 @@
+<?xml version='1.0' encoding='UTF-8'?>
```
Comment (amstone326) [inline app/unit-tests/resources/commcare-apps/phone_call/media_suite.xml]: Think you can also get rid of this

**Comment by wpride** [comment]
@amstone326 thanks for the feedback, updated. will add docs

**Pull request merged**